### PR TITLE
Add helpers allowing to check whether the method is a nonmutating operator method or a nonmutating method

### PR DESCRIPTION
**Pull request by fatkodima**
Moved from https://github.com/rubocop-hq/rubocop/pull/8155

For context https://github.com/rubocop-hq/rubocop-performance/pull/140/files#r439732580

Bullet number 4

### CODE REVIEW

File: lib/rubocop/ast/node/mixin/method_identifier_predicates.rb [hunk 1]
```diff
@@ -1,22 +1,69 @@
 # frozen_string_literal: true
 
+require 'set'
```
Comment (marcandre) [inline lib/rubocop/ast/node/mixin/method_identifier_predicates.rb:3]: The gem `parser` (and maybe others) already `require 'set'`. I'd remove it entirely, but if you prefer having it for clarity, it belongs in `lib/reubocop/ast.rb`.

File: spec/rubocop/ast/send_node_spec.rb [hunk 2]
```diff
@@ -736,6 +736,126 @@ module Foo
     end
   end
 
+  describe '#nonmutating_binary_operator_method?' do
+    context 'with a nonmutating binary operator method' do
+      let(:source) { 'foo + bar' }
+
+      it { expect(send_node.nonmutating_binary_operator_method?).to be_truthy }
```
Comment (marcandre) [inline spec/rubocop/ast/send_node_spec.rb:743]: Please use `be(true)` and `be(false)` until we can do better

File: lib/rubocop/ast/node/mixin/method_identifier_predicates.rb (hunk 1 above)
Comment (fatkodima) [reply lib/rubocop/ast/node/mixin/method_identifier_predicates.rb]: > but if you prefer having it for clarity

yes, this. 
And imo it is better to not depend on something else required it.
Moved to suggested location.

File: spec/rubocop/ast/send_node_spec.rb (hunk 2 above)
Comment (fatkodima) [reply spec/rubocop/ast/send_node_spec.rb:743]: Yes, the whole test codebase is not written the way I would prefer, but I was consistent with what we have 😄

**Pull request merged**